logical/plugin/plugin.go

```diff
@@ -96,9 +96,18 @@ func NewBackend(ctx context.Context, pluginName string, sys pluginutil.LookRunne
 
 func newPluginClient(ctx context.Context, sys pluginutil.RunnerUtil, pluginRunner *pluginutil.PluginRunner, logger log.Logger, isMetadataMode bool) (logical.Backend, error) {
 	// pluginMap is the map of plugins we can dispense.
-	pluginMap := map[string]plugin.Plugin{
-		"backend": &BackendPlugin{
-			metadataMode: isMetadataMode,
+	pluginSet := map[int]plugin.PluginSet{
```

Comment:
Is it possible to abstract out the creation of PluginSet into its own function, probably parameterized by `isMetadataMode`? An addition to this set currently requires updates in multiple places.

Also, probably use consts for version numbers?

Reply:
Each set is subtly different, it depends on client/server and each have different requirements